pkg/provisioner/ironic/ironic.go

```diff
@@ -461,9 +471,22 @@ func setDeployImage(config ironicConfig, accessDetails bmc.AccessDetails, hostIm
 				deployImageInfo[deployKernelKey] = config.deployKernelURL
 			}
 			deployImageInfo[deployRamdiskKey] = hostImage.ImageURL
+			// Extra params from the kernel have been already posted to
+			// Ironic node API in the Register function, so extra kernel args in driver info need
+			// update only if the dynamically generated preprov image requires it
 			if hostImage.ExtraKernelParams != "" {
-				// Using %default% prevents overriding the config in ironic-image
-				deployImageInfo[kernelParamsKey] = "%default% " + hostImage.ExtraKernelParams
+				// Combine the default ironic node kernel params with the extra
+				// params from the BMH and the dynamically generated preprov image
+				comExtraKernelParams := defaultKernelParam
+				trimDataKParams := strings.TrimSpace(data.PreprovisioningExtraKernelParams)
+				trimPreprovImgKParams := strings.TrimSpace(hostImage.ExtraKernelParams)
+				if trimDataKParams != "" {
+					comExtraKernelParams += " " + data.PreprovisioningExtraKernelParams
+				}
+				if trimPreprovImgKParams != "" {
+					comExtraKernelParams += " " + hostImage.ExtraKernelParams
```

Comment:
Here you are adding the kargs supplied by the PreprovisioningImage to the ones passed in the BMH Spec.
But you already did that in the BMH controller (which is the correct place to do it - this shouldn't be provisioner-specific). So the ones from the PreprovisioningImage will be duplicated.